### PR TITLE
Task marking and clearing completed task

### PR DESCRIPTION
**Pull request by rolandfbk**
Below are the features that I have implemented:

- The user can select the completed tasks
- The user can select the completed tasks and not delete them. On a new page load, the user can still see the checked completed task and decide to clear them or not.
- The user can select the completed tasks, then click on "Clear all completed" to delete them

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import './style.css';
+import { taskMarking, clearCompletedtask } from './task-mark-module.js';
+// import { clearCompletedtask } from './task-mark-module.js';
```
Comment (hiromataba) [inline src/app.js]: - [ ] Please remove commented code to improve the quality of your code and make it cleaner

File: src/task-mark-module.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const clearTaskListSet = JSON.parse(JSON.stringify(localStorage.getItem('taskList')));
+    const newClearTaskListObjSet = JSON.parse(clearTaskListSet);
+    const newArray = [];
+
+    for (let i = 0; i < newClearTaskListObjSet.length; i += 1) {
+      if (newClearTaskListObjSet[i].completed === false) {
+        newArray.push(newClearTaskListObjSet[i]);
+      }
+    }
+
```
Comment (hiromataba) [inline src/task-mark-module.js]: - [ ] To clear all completed tasks you were required to use the (filter) method, `Array.filter()` :point_down: 

image

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @rolandfbk    :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elyor-doniyorov — APPROVED** [review]
Hi @rolandfbk, 

In my opinion, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**